codegen/grid.py

```diff
@@ -717,6 +720,13 @@ def set_variable(self, var, value=0, type='int', constant=False):
             var = var.name
         self.defined_variable[var] = Variable(var, value, type, constant)
 
+    def set_page_size(self, page_size):
```

Comment:
- function name changed to set_alignment
- added parameter int size to int opesci_read_simple_binary_ptr(), compare size with file size when reading data
- having googled a bit I'm not sure how to allocate dynamic memory using __attribute__, apart from using global variable. These guys seem to be using posix:
http://www.prace-ri.eu/IMG/pdf/Best-Practice-Guide-Intel-Xeon-Phi.pdf